tox.ini

```diff
@@ -71,6 +71,10 @@ commands =
          coverage: coverage xml
          coverage: coverage report
 
+         # Also run tests with -bb to catch bugs like `"%s" % (some_bytes,)`.
+         # Eventually everything should run with this, but so far only fixed
+         # some of the Python3-ported modules.
+         python -bb -m twisted.trial --rterrors allmydata.test.web
```

Comment:
Cool.

I'm thinking about how we can scale this check up further.  Passing with "-bb" is related to but separate from "passes on Python 3" right?  So we can't just split the `PORTED_TEST_MODULES` list into two pieces.  Would it make sense to have a separate `BB_TEST_MODULES` list where test modules that pass w/ -bb get added?  

What might really be nice is if we could run each test just once in the strictest configuration it is known to pass.  That is, create a `BB_TEST_MODULES` list and subtract it from `TEST_SUITE`.  That suddenly creates a lot more complexity though :(

Maybe this is fine as-is for now.  We hope to have someone joining soon who will focus on testing improvements.  One such improvement could be sorting out this mess - or implementing some plan, if you feel like you have an idea of what might make sense to implement.  If so, please write it up in a ticket and link to it from here.

Reply:
My experience was that fixing -bb for web wasn't _that_ much work, so might be able to just... enable it for all Python 3 tests withouts too much effort in one PR. So I filed ticket for that and then try that:

https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3618